### PR TITLE
CORTX-31357 Modified TEST-5513 and TEST-5516 as per review comment

### PR DESCRIPTION
**Pull request by AshwiniS-Borse**
Signed-off-by: AshwiniS-Borse <ashwini.borse@seagate.com>

# Problem Statement
Jira ticket - CORTX-31357
- Modified TEST-5513 and TEST-5516 as per review comment 

# Design
-  [(https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/986808747/F-28A+TDS+-+S3+Bucket+Tagging+API)]

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [x] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)
          - Pytest Collection is green

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [x] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [x] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by AshwiniS-Borse** [comment]
@apoorva-rao @archanabpatil @VikasKumarC @DhobaleVishal Please review.
Changes are made as per Chetan's comment mentioned in confluence.
https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/986808747/F-28A+TDS+-+S3+Bucket+Tagging+API
`Why set needed, can do Get directly on non existing bucket rt ? as test is about get on non-existing bucket.

But yes another perspective is that put on non-existing bucket may fail, but still better to get it again, to make sure, put & get has same results. so put & get steps can in previous test case itself, while get test case here can limit to get alone`

**Review by apoorva-rao — APPROVED** [review]

File: tests/s3/test_bucket_tagging.py [hunk 1]
```diff
@@ -757,6 +757,16 @@ def test_2451(self):
...
+        self.log.info("Step 2: Retrieving tag of non existing bucket")
+        try:
+            resp = self.tag_obj.get_bucket_tags(self.bucket_name)
+            self.log.info(resp)
+            assert_utils.assert_false(resp[0], resp[1])
+        except CTException as error:
+            self.log.info(error)
+            assert_utils.assert_in(errmsg.NO_BUCKET_OBJ_ERR_KEY, str(error.message), error.message)
+        self.log.info("Step 2: Retrieved tag of non existing bucket failed with NoSuchBucket")
+        self.log.info("Step 2: Verified PUT and GET tag of non existing bucket failed with NoSuchBucket")
```
Comment (apoorva-rao) [inline tests/s3/test_bucket_tagging.py:769]: Please fix the codacy issue

**Review by apoorva-rao — CHANGES_REQUESTED** [review]
@AshwiniS-Borse Please see inline comment for the codacy issue that needs fixing

**Review by apoorva-rao — APPROVED** [review]

File: tests/s3/test_bucket_tagging.py [hunk 2]
```diff
@@ -757,6 +757,17 @@ def test_2451(self):
             self.log.info(error)
             assert_utils.assert_in(errmsg.NO_BUCKET_OBJ_ERR_KEY, str(error.message), error.message)
         self.log.info("Step 1: Setting a tag for non existing bucket failed with: NoSuchBucket")
+        self.log.info("Step 2: Retrieving tag of non existing bucket")
+        try:
+            resp = self.tag_obj.get_bucket_tags(self.bucket_name)
+            self.log.info(resp)
```
Comment (DhobaleVishal) [inline tests/s3/test_bucket_tagging.py:763]: Seems repeated logging statement, as we are logging it inside function.

File: tests/s3/test_bucket_tagging.py [hunk 3]
```diff
@@ -757,6 +757,17 @@ def test_2451(self):
...
+            assert_utils.assert_false(resp[0], resp[1])
+        except CTException as error:
+            self.log.info(error)
```
Comment (DhobaleVishal) [inline tests/s3/test_bucket_tagging.py:766]:
```suggestion
            self.log.exception(error)
```

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]

File: tests/s3/test_bucket_tagging.py (hunk 3 above)
Comment (AshwiniS-Borse) [reply tests/s3/test_bucket_tagging.py]: Updated changes and validated against new code changes

**Review by DhobaleVishal — APPROVED** [review]

**Review by VikasKumarC — APPROVED** [review]

**Review by seagate-sarang-sawant — APPROVED** [review]

**Pull request merged**